src/main/scala/mesosphere/marathon/tasks/TaskTracker.scala

```diff
@@ -1,259 +1,32 @@
-    cachedApps.getOrElseUpdate(appId, fetchApp(appId))
-  }
-
-  // FIXME: remove, see documented deficiencies above
-  def list: Map[PathId, App] = cachedApps.mapValues(_.toApp).toMap
-
-  def count(appId: PathId): Int = getTaskMap(appId).size
-
-  // FIXME: remove, see documented deficiencies above
-  def contains(appId: PathId): Boolean = cachedApps.contains(appId)
-
-  def take(appId: PathId, n: Int): Set[MarathonTask] = getTasks(appId).iterator.take(n).toSet
-
-  /**
-    * Create a new task and store it in-memory and in the underlying storage.
-    *
-    * If the task exists already, the returned Future will fail.
-    */
-  def created(appId: PathId, task: MarathonTask): Future[MarathonTask] = {
-    val taskId = task.getId
-    getTaskMap(appId).putIfAbsent(task.getId, task) match {
-      case Some(existingTask) =>
-        Future.failed(new IllegalStateException(s"cannot create task [$taskId] of app [$appId], it already exists"))
-      case None =>
-        repo.store(task).map { task =>
-          log.debug(s"created task [$taskId] of app [$appId]")
-          task
-        }
-    }
-  }
-
-  /**
-    * Process a status update for an existing task and either updates the tasks or removes
-    * it from the TaskTracker.
-    *
-    * If the task does not exist yet, the returned Future will fail.
-    */
-  def statusUpdate(appId: PathId, status: TaskStatus): Future[_] = {
-    val taskId = status.getTaskId.getValue
-    val app = getInternalApp(appId)
-    val maybeAppTask: Option[MarathonTask] = app.tasks.get(taskId)
-
-    maybeAppTask match {
-      case Some(task) =>
-        processUpdate(app, task, status)
-      case None =>
-        Future.failed(new IllegalStateException(s"task [$taskId] of app [$appId] does not exist"))
-    }
-  }
+trait TaskTracker {
 
-  private[this] def processUpdate(app: InternalApp, task: MarathonTask, status: TaskStatus): Future[_] = {
-    def updateTaskOnStateChange(task: MarathonTask): Future[_] = {
-      def statusDidChange(statusA: TaskStatus, statusB: TaskStatus): Boolean = {
-        val healthy = statusB.hasHealthy &&
-          (!statusA.hasHealthy || statusA.getHealthy != statusB.getHealthy)
+  def getTasks(appId: PathId): Iterable[MarathonTask]
 
-        healthy || statusA.getState != statusB.getState
-      }
+  def getTask(appId: PathId, taskId: String): Option[MarathonTask]
 
-      if (statusDidChange(task.getStatus, status)) {
-        updateTask(task.toBuilder.setStatus(status).build)
-      }
-      else {
-        log.debug(s"Ignoring status update for ${task.getId}. Status did not change.")
-        Future.successful(())
-      }
-    }
+  def getVersion(appId: PathId, taskId: String): Option[Timestamp]
 
-    def updateTask(updatedTask: MarathonTask): Future[_] = {
-      val appId = app.appName
-      val taskId: String = updatedTask.getId
-      log.debug(s"updating task [$taskId] of app [$appId]: {}", updatedTask)
+  def list: Map[PathId, TaskTracker.App]
```

Comment:
Well, the comment says:

  // FIXME: remove, see documented deficiencies above

So without any context, I assume that the complete method should be removed, which would implied that it will also be removed from the trait and should be used in new code.